### PR TITLE
More use of Optional and some other fixes

### PR DESCRIPTION
**Pull request by oscargus**
* Used `Optional` instead of `return null`. Sometimes it makes much sense, sometimes less.
* Replaced checking zero or non-zero `String` lengths with `.isEmpty()`
* Changed the default case in some switch-statements to avoid `return null` (as that case should never happen anyway)
* Avoided an exception from happening when importing items without author and generating a BibTexKey which is based on the author name.

### CODE REVIEW

File: src/main/java/net/sf/jabref/JabRef.java [hunk 1]
```diff
@@ -898,11 +889,11 @@ private static ParserResult importFile(String argument) {
...
-    private static ParserResult importToOpenBase(String argument) {
-        ParserResult result = JabRef.importFile(argument);
+    private static Optional<ParserResult> importToOpenBase(String argument) {
+        Optional<ParserResult> result = JabRef.importFile(argument);
 
-        if (result != null) {
-            result.setToOpenTab(true);
+        if (result.isPresent()) {
+            result.get().setToOpenTab(true);
         }
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/JabRef.java]: in that case, use the following:

```
optional.ifPresent(value -> value.setToOpenTab(true));
```

The function is only executed if the optional is present.

File: src/main/java/net/sf/jabref/external/SynchronizeFileField.java [hunk 2]
```diff
@@ -123,16 +123,16 @@ public void run() {
...
+            
                 Object old = sel[i].getField(fieldName);
                 FileListTableModel tableModel = new FileListTableModel();
                 if (old != null)
                     tableModel.setContent((String)old);
                 Thread t = FileListEditor.autoSetLinks(sel[i], tableModel, null, null);
-
+            
                 if (!tableModel.getStringRepresentation().equals(old)) {
                     String toSet = tableModel.getStringRepresentation();
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/external/SynchronizeFileField.java]: Just delete commented out code.

File: src/main/java/net/sf/jabref/logic/labelPattern/LabelPatternUtil.java [hunk 3]
```diff
@@ -175,8 +175,11 @@ private static String unifyDiacritics(String content) {
      * @return True if the author or editor is an institution.
      */
     private static boolean isInstitution(String author) {
-        return (author.charAt(0) == '{')
-                && (author.charAt(author.length() - 1) == '}');
+        if (!(author.isEmpty())) {
+            return (author.charAt(0) == '{') && (author.charAt(author.length() - 1) == '}');
+        } else {
+            return false; // In case of empty or null author
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/logic/labelPattern/LabelPatternUtil.java]: if author is null, a NPE will be thrown.

File: src/main/java/net/sf/jabref/JabRef.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/JabRef.java]: Yes, there are a few more cases like that. I'll change them (now that I learned more about how to use Optional, didn't want to update during possible review...).

File: src/main/java/net/sf/jabref/logic/labelPattern/LabelPatternUtil.java (hunk 3 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/labelPattern/LabelPatternUtil.java]: Yes, bad comment. I had null check earlier, but then I realized that the method is never(?) called with a null argument anyway. Let me double check and come back with the correct solution. :-)

Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/labelPattern/LabelPatternUtil.java]: As this method will at worst be called with an empty String, I just changed the comment. Also, the same holds for the methods `removeDiacritics()` and `generateInstitutionKey()` so I changed the null test to `isEmpty()` there.

**Pull request merged**